pkgs/applications/networking/nextcloud-client-bin/generic.nix

```diff
@@ -0,0 +1,50 @@
+{ darwin
+, p7zip
+, libarchive
+, stdenv
+, lib
+, fetchurl
+}:
+{ pname, version, url, hash }:
+
+let
+  inherit (stdenv) targetPlatform;
+  ARCH = if targetPlatform.isArch64 then "arm64" else "x64";
+in stdenv.mkDerivation rec {
+  inherit pname version;
+
+  src = fetchurl {
+    inherit url hash;
+  };
+
+  unpackPhase = lib.optionalString stdenv.isDarwin
+  ''
+    7z x $src
+    bsdtar -xf Payload~
+  '';
+
+  nativeBuildInputs = lib.optionalString stdenv.isDarwin [
+    p7zip
+    libarchive
+  ];
+
+  installPhase = lib.optionalString stdenv.isDarwin
```

Comment:
There is no need to use `lib.optionalString stdenv.isDarwin` as the package can only be build on Darwin

Reply:
I figured it would be logical to keep it open for the future in case someone(Or even I) wants to add Linux support. Having seen derivations specifically made with Linux users in mind, I figured this more system agnostic change would be beneficial in the long run

Reply:
> I figured it would be logical to keep it open for the future in case someone(Or even I) wants to add Linux support. Having seen derivations specifically made with Linux users in mind, I figured this more system agnostic approach would be beneficial in the long run

That's great, but as of now, according to the `meta`, this derivation can only be built on Darwin. So, these checks are just useless overhead at the moment. You or someone else can add them again once both are supported.

Reply:
Ok removed the bits and resquashed the commit